src/uhs/transaction/wallet.cpp

```diff
@@ -121,6 +154,35 @@ namespace cbdc {
         return ret;
     }
 
+    auto transaction::wallet::generate_minter_key() -> pubkey_t {
+        if(m_pubkeys.empty()) {
+            // new wallet. create the minter key
+            return generate_key();
+        }
+        // first key is always the minter key
+        return m_pubkeys[0];
+    }
+
+    auto transaction::wallet::generate_test_minter_key() -> pubkey_t {
```

Comment:
I'm confused about this. Is the test minter key the one the validation code will check signed the TX? How is that different from the key generated when I can `generate_minter_key()`?

What if I call `generate_minter_key()` first, and then `generate_test_minter_key()` after? Now `m_pubkeys[1]` contains the minter key. Same thing if I call `generate_test_minter_key()` multiple times.

Reply:
`generate_test_minter_key` generates a pre-determined key that's preset in all existing `.cfg` files for testing and/or demos.  Again, I was looking for a way to minimize changes across the codebase. See https://github.com/mit-dci/opencbdc-tx/blob/eb16c651e649042ddbbfbea3b05eb0b219dae981/src/uhs/transaction/wallet.hpp#L129 for a descriptoin

Reply:
Can we not read the key from the config file? Why do we need code to re-generate it?

Reply:
@metalicjames  Mainly because tests call wallet code to mint (especially the integration tests) and you need a deterministic key for that.  Putting the minter's private key in the configuration file for testing would mean you should put all minter private keys in the configuration file (testing and non-testing) to keep the design consistent.   But IMO, putting minter (or any) private keys in the configuration file seems like a really bad idea from a security perspective.

Reply:
Worth noting that this is not a production system, so we don't have to be quite as worried about such things at the moment.

For example, for sentinel attestation checks, the sentinel private keys *are* stored in the configuration files.

Reply:
@HalosGhost Agree and I understand what you're saying.  But on the other hand, adding security late to the process usually doesn't turn out well.   If security is not a high-priority in the current context of the project, then is secure minting even necessary right now?  Just thinking out loud on how to best bound the problem...

Reply:
I take your point. I suppose my view is that we should try to scope our questions as small as possible.

I think this PR is answering the question “How could secure minting be supported?” That doesn't require all other best-practices to be addressed at the same time. To that end, having the keys in-configuration or not is an orthogonal concern (in my opinion).